src/App.tsx

```diff
@@ -1,26 +1,273 @@
 /* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+/* eslint-disable jsx-a11y/label-has-associated-control */
 
-const USER_ID = 0;
+import React, { useState, useRef, useEffect } from 'react';
+import * as todosServise from './api/todos';
+import { UserWarning } from './UserWarning';
+import { Todo, TodoTitleOrCompleted } from './types/Todo';
+import { SelectedBy } from './types/SelectedBy';
+import Header from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import ErrorNotification from './components/ErrorNotification';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedBy, setSelectedBy] = useState<SelectedBy>(SelectedBy.all);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isDisabletField, setIsDisabletField] = useState(false);
+  const [title, setTitle] = useState('');
+  const mainInputRef = useRef<HTMLInputElement>(null);
+
+  const [loadingTodos, setLoadingTodos] = useState<Record<number, boolean>>({});
+
+  // #region loading todos
+  useEffect(() => {
+    mainInputRef.current?.focus();
+    todosServise
+      .getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+        setTimeout(() => setErrorMessage(''), 3000);
+      });
+  }, []);
+
+  if (!todosServise.USER_ID) {
     return <UserWarning />;
   }
+  // #endregion
+
+  // #region create Todo
+  const createTodo = () => {
+    if (!title.trim()) {
+      setErrorMessage('Title should not be empty');
+      setTimeout(() => setErrorMessage(''), 3000);
+
+      return;
+    }
+
+    setIsDisabletField(true);
+
+    setTempTodo({
+      id: 0,
+      title,
+      userId: todosServise.USER_ID,
+      completed: false,
+    });
+
+    return todosServise
+      .createTodos({
+        title: title.trim(),
+        userId: todosServise.USER_ID,
+        completed: false,
+      })
+      .then(result => {
+        setTodos(prev => [...prev, result]);
+        setTitle('');
+      })
+      .catch(() => {
+        setErrorMessage('Unable to add a todo');
+        setTimeout(() => setErrorMessage(''), 3000);
+      })
+      .finally(() => {
+        setTempTodo(null);
+        setIsDisabletField(false);
+        setTimeout(() => {
+          mainInputRef.current?.focus();
+        }, 300);
+      });
+  };
+  // #endregion
+
+  // #region edit Todo
+  function editTodo(data: TodoTitleOrCompleted, todoId: number) {
```

Comment:
It is important to follow the same way of creating functions everywhere, fix it everywhere
```suggestion
  сщтіе editTodo = (data: TodoTitleOrCompleted, todoId: number) => {
```